nemo/collections/tts/modules/fastpitch.py

```diff
@@ -262,7 +274,10 @@ def forward(
             len_regulated, dec_lens = regulate_len(durs_predicted, enc_out, pace)
 
         # Output FFT
-        dec_out, _ = self.decoder(input=len_regulated, seq_lens=dec_lens)
+        if self.speaker_emb_condition_decoder:
+            dec_out, _ = self.decoder(input=len_regulated, seq_lens=dec_lens, conditioning=spk_emb)
+        else:
+            dec_out, _ = self.decoder(input=len_regulated, seq_lens=dec_lens)
```

Comment:
Can similarly reduce duplication like:

```
if self.speaker_emb_condition_decoder:
  dur_conditioning = spk_emb
else:
  dur_conditioning  = 0.0
dec_out, _ = self.decoder(input=len_regulated, seq_lens=dec_lens, conditioning=spk_emb)
```

Reply:
I believe the current implementation is intuitive in terms of readability. Like we can see that there is no conditioning in one case and there is conditioning in another. Whereas it might be a little confusing as in what is 0.0 conditioning. This is what I think, let me know if that makes sense.

Reply:
Setting it to 0.0 is confusing. Leaving it as is would be alright. Some alternatives would be:

(a) Set conditioning = None and then change the transformer logic to handle the None input.
(b) Make the behavior consistent with the pitch/duration predictor by removing "conditioning" argument and setting input = len_regulated or len_regulated + spk_emb